bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/nhc2/NhcMqttConnection2.java

```diff
@@ -368,6 +368,6 @@ public void onSuccess(String topic) {
 
     @Override
     public void onFailure(String topic, Throwable error) {
-        logger.debug("Niko Home Control: publish failed {}, {}", topic, error);
+        logger.debug("Niko Home Control: publish failed {}, {}", topic, error.getMessage());
```

Comment:
If you catch all `Exception` you are potentially also catching for example `NullPointerException`s and  they tend to miss a message, so you will be printing the message `null` which will make it troublesome to debug, in this case it is always better to log the complete stacktrace.

Reply:
This is the only one that might be new :-)

Reply:
I re-added that, but I guess NPE or similar will not be passed as `error` to that method.